dynamiqs/mesolve/mesolve.py

```diff
@@ -167,6 +162,8 @@ def broadcast(x: TimeArray) -> TimeArray:
         f = _cartesian_vectorize(_mesolve, n_batch, out_axes)
     else:
         f = _flat_vectorize(_mesolve, n_batch, out_axes)
+        H = squeeze_ones(H)
+        jump_ops = list(map(squeeze_ones, jump_ops))
```

Comment:
If the passed Hamiltonian is of shape `(1, n, n)` or `(2, 1, n, n)` we don't want to squeeze the one. IMO `flat_vectorize` the magic you propose should be done by `_flat_vectorize`, it has enough information with the shape of the objects.

Reply:
You have to squeeze for emulating the broadcasting. I think it works as is but I'll add a test for your edge case.

Reply:
But you're also going to squeeze the original dimensions equal to one no?

Reply:
Rho is not squeezed so shapes may be preserved